adapter/pkg/messaging/azure_connection.go

```diff
@@ -37,45 +48,175 @@ func init() {
 	AzureNotificationChannel = make(chan []byte)
 }
 
-// InitiateBrokerConnectionAndGetAvailableTopics to initiate connection and get topic list
-func InitiateBrokerConnectionAndGetAvailableTopics(eventListeningEndpoint string) (*servicebus.Namespace,
-	[]*servicebus.TopicEntity, error) {
-	var err error
-	var availableTopics []*servicebus.TopicEntity
+// InitiateBrokerConnectionAndValidate to initiate connection and validate azure service bus constructs to
+// further process
+func InitiateBrokerConnectionAndValidate(eventListeningEndpoint string, componentName string, reconnectRetryCount int,
+	reconnectInterval time.Duration, subscriptionIdleTimeDuration time.Duration) ([]SubscriptionType, error) {
 	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Trying to connect to azure service bus with " +
 		"connection string " + eventListeningEndpoint)
+	subscriptionMetaDataList := make([]SubscriptionType, 0)
 	namespace, err := servicebus.NewNamespace(servicebus.NamespaceWithConnectionString(eventListeningEndpoint))
-
 	if err == nil {
+		var getTopicListError error
+		var subscriptionMetaDataCreationError error
 		logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Successfully received namespace ")
 		topicManager := namespace.NewTopicManager()
-		ctx, cancel := context.WithCancel(context.Background())
-		defer cancel()
-		availableTopics, err = topicManager.List(ctx)
-		//TODO (dnwick) (add retry logic)
-		if err != nil {
-			logger.LoggerMgw.Errorf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying to get topic " +
-				"list from azure service bus :%v", err)
+		var availableTopics []*servicebus.TopicEntity
+		for j := 0; j < reconnectRetryCount; j++ {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			availableTopics, getTopicListError = topicManager.List(ctx)
+			if getTopicListError != nil {
+				logger.LoggerMgw.Errorf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying to get topic "+
+					"list from azure service bus :%v. Retrying after %d seconds", err, reconnectInterval)
```

Comment:
Let's log the retry attempt if it's greater than 0

Reply:
done